src/rai_core/rai/state_based/ros2/aggregators.py

```
@@ -0,0 +1,166 @@
+        return f"{self.__class__.__name__}(source={self.source}"
+
+
+class Ros2LogsAggregator(BaseStateAggregator):
+    """Returns only unique messages while keeping their order"""
+
+    levels = {10: "DEBUG", 20: "INFO", 30: "WARNING", 40: "ERROR", 50: "FATAL"}
+
+    def __init__(
+        self,
+        source: str = "/rosout",
+        topic_config: TopicConfig = TopicConfig(msg_type="rcl_interfaces/msg/Log"),
```

Ruff (0.8.2)

Do not perform function call `TopicConfig` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)